x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/scalar/ScalarFunction.java

```diff
@@ -132,9 +134,21 @@ private String basicTemplate(Function function) {
             (function instanceof AggregateFunction && ((AggregateFunction) function).field().dataType() == DATETIME)) {
 
             return "{sql}.asDateTime({})";
-        } else {
-            return "{}";
+        } else if (function instanceof AggregateFunction) {
```

Comment:
Same here: please, let's have a comment for the future to show that this part with the cast is needed for the bucket selector so the case of `HAVING`.

Reply:
Sure, added.